text/0000-ghost-busting.md

```diff
@@ -0,0 +1,732 @@
+- Feature Name: `ghost_busting`
+- Start Date: 2018-03-04
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+[`PhantomData<T>`]: https://doc.rust-lang.org/nightly/nomicon/phantom-data.html
+[variance]: https://doc.rust-lang.org/nightly/nomicon/subtyping.html
+[drop checking]: https://doc.rust-lang.org/nightly/nomicon/dropck.html
+
+1. Defines unused type parameters to act as if they are logically owned
+   and covariant.
+
+2. Introduces `phantom T` pseudo-fields which improves the ergonomics of
+   changing [variance] and [drop checking] behavior when it is needed.
```

Comment:
And auto trait implementations?

Reply:
Elaborate?

Reply:
`PhantomData<T>` only implements auto traits that `T` does. This is used in [fmt](https://github.com/rust-lang/rust/blob/master/src/libcore/fmt/mod.rs#L265) to opt out of all auto traits.

Reply:
Today I learned... Interesting. I'll write something about it when I'm more lucid =P

Reply:
I've updated the RFC accordingly now.